docs/understanding-docker.md

```diff
@@ -17,273 +17,282 @@ weight = -90
+set of steps we call *instructions*, which are stored in a `Dockerfile`. Each
+instruction creates a new layer in the image. Some examples of Dockerfile
+instructions are:
 
-* Run a command
-* Add a file or directory
-* Create an environment variable
-* What process to run when launching a container from this image
+* Specify the base image (`FROM`)
+* Specify the maintainer (`MAINTAINER`)
+* Run a command (`RUN`)
+* Add a file or directory (`ADD`)
+* Create an environment variable (`ENV`)
+* What process to run when launching a container from this image (`CMD`)
 
-These instructions are stored in a file called a `Dockerfile`. A `Dockerfile` is
-a text based script that contains instructions and commands for building the image
-from the base image. Docker reads this `Dockerfile` when you request a build of
-an image, executes the instructions, and returns a final image.
+Docker reads this `Dockerfile` when you request a build of
+an image, executes the instructions, and returns the image.
 
 ### How does a Docker registry work?
-The Docker registry is the store for your Docker images. Once you build a Docker
-image you can *push* it to a public registry such as [Docker Hub](https://hub.docker.com)
-or to your own registry running behind your firewall.
-
-Using the Docker client, you can search for already published images and then
-pull them down to your Docker host to build containers from them.
-
-[Docker Hub](https://hub.docker.com) provides both public and private storage
-for images. Public storage is searchable and can be downloaded by anyone.
-Private storage is excluded from search results and only you and your users can
-pull images down and use them to build containers. You can [sign up for a storage plan
-here](https://www.docker.com/pricing).
+A Docker registry stores Docker images. After you build a Docker image, you
+can *push* it to a public registry such as [Docker Hub](https://hub.docker.com)
+or to a private registry running behind your firewall. You can also search for
+existing images and pull them from the registry to a host.
+
+[Docker Hub](https://hub.docker.com) is a hosted Docker registry which provides
+both public and private storage locations (repositories) for images. Images
+stored in a public repository on Docker Hub can be searched for and downloaded
+by any user. Private repositories are excluded from search results and only that
+repository's users can pull images down and use them to build containers. You can
+[sign up for a repository plan here](https://www.docker.com/pricing).
 
 ### How does a container work?
-A container consists of an operating system, user-added files, and meta-data. As
-we've seen, each container is built from an image. That image tells Docker
-what the container holds, what process to run when the container is launched, and
-a variety of other configuration data. The Docker image is read-only. When
+A container consists of an operating system, any extra files you add at image
```

Comment:
A container consists of a Linux kernel (shared across containers), a filesystem, and an application or piece of software with all its dependencies: code, runtime, tools, system libraries.